foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/widget/WidgetVisitorUtility.java

```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+  * Copyright (c) 2016 Red Hat, Inc.
+  * Distributed under license by Red Hat, Inc. All rights reserved.
+  * This program is made available under the terms of the
+  * Eclipse Public License v1.0 which accompanies this distribution,
+  * and is available at http://www.eclipse.org/legal/epl-v10.html
+  *
+  * Contributor:
+  *     Red Hat, Inc. - initial API and implementation
+  ******************************************************************************/
+package org.jboss.tools.foundation.ui.widget;
+
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+
+/**
+ * A utility for all tasks that may require logic for visiting controls,
+ * composites, or other widgets. 
+ */
+public class WidgetVisitorUtility {
+
+	/**
+	 * Recursively enable all children of this composite
+	 * @param composite
+	 */
+	public void enableAllChildren(Composite composite) {
+		enableAllChildren(true, composite);
+	}
+	
+	/**
+	 * Recursively disable all children of this composite
+	 * @param composite
+	 */
+	public void disableAllChildren(Composite composite) {
+		enableAllChildren(false, composite);
+	}
+	
+	/**
+	 * Optionally enable or disable all children (recursive) of this composite
+	 * @param enabled
+	 * @param composite
+	 */
+	public void enableAllChildren(final boolean enabled, final Composite composite) {
+		doForAllChildren(new IWidgetVisitor() {
+			@Override
+			public boolean visit(Control control) {
+				control.setEnabled(enabled);
+				return true;
+			}
+		}, composite);
+	}
+	
+	/**
+	 * Functionally equivalent to accept(visitor, composite, false, false)
+	 * for a non-depth-first recursive call that ignores the parent composite itself. 
+	 * @param visitor
+	 * @param composite
+	 */
+	public void doForAllChildren(IWidgetVisitor visitor, Composite composite) {
+		accept(visitor, composite, false, false);
+	}
```

Comment:
I would invert the parameters so that it'd be more natural to write 
```java
doForAllChildren( composite, control -> {
        control.setEnabled(enabled);
 	return true;
});
```